### PR TITLE
Fix cli blockingioerror

### PR DESCRIPTION
**Pull request by trammell**
**Proposed changes**:

Add `print_blocking()` and try/except block to manage blocking print errors in CLI.

Fixes https://rasa-open-source.atlassian.net/browse/OSS-183

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by AlvaroLeles — APPROVED** [review]

File: rasa/shared/utils/cli.py [hunk 1]
```diff
@@ -14,7 +26,10 @@ def print_color(*args: Any, color: Text) -> None:
...
         print(output, file=stream)
     except ImportError:
-        print(output)
+        try:
+            print(output)
+        except BlockingIOError:
+            print_blocking(output)
 
 
 def print_success(*args: Any) -> None:
```
Comment (IanFPFerreira) [inline rasa/shared/utils/cli.py:35]: """Prints the given arguments in green color, indicating success.

    Args:
        *args: Any type of arguments to be printed.
    """
Add docstring

**Review by IanFPFerreira — CHANGES_REQUESTED** [review]

File: rasa/shared/utils/cli.py (hunk 1 above)
Comment (trammell) [reply rasa/shared/utils/cli.py:35]: This docstring is correct and valid but has nothing to do with the defect or the PR, so it doesn't belong here.

File: rasa/shared/utils/cli.py [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+    """Saves fcntl settings and restores them after print."""
+    save = fcntl.fcntl(sys.stdout.fileno(), fcntl.F_GETFL)
+    new = save & ~os.O_NONBLOCK
+    fcntl.fcntl(sys.stdout.fileno(), fcntl.F_SETFL, new)
+    print(string)
+    fcntl.fcntl(sys.stdout.fileno(), fcntl.F_SETFL, save)
+    sys.stdout.flush()
+
+
 def print_color(*args: Any, color: Text) -> None:
```
[the remark this answers is not in the document]
Comment (trammell) [reply rasa/shared/utils/cli.py:19]: Changing "Prints" to "Print" for imperative mood as per https://peps.python.org/pep-0257/

File: rasa/shared/utils/cli.py [hunk 3]
```diff
@@ -1,10 +1,28 @@
+import fcntl
+import os
 import sys
 from typing import Any, Text, NoReturn
 
 import rasa.shared.utils.io
 
 
+def print_blocking(string) -> None:
+    """Save fcntl settings and restore them after print()."""
```
Comment (ancalita) [inline rasa/shared/utils/cli.py:10]: I'm unfamiliar with `fcntl`, could we expand the docstring a bit more about the need for this.

File: rasa/shared/utils/cli.py [hunk 4]
```diff
@@ -1,10 +1,28 @@
...
+    save = fcntl.fcntl(sys.stdout.fileno(), fcntl.F_GETFL)
+    new = save & ~os.O_NONBLOCK
```
Comment (ancalita) [inline rasa/shared/utils/cli.py:12]: could we add some comment above this line to explain what it's doing concisely.

File: rasa/shared/utils/cli.py [hunk 5]
```diff
@@ -1,10 +1,28 @@
+import fcntl
+import os
 import sys
 from typing import Any, Text, NoReturn
 
 import rasa.shared.utils.io
 
 
+def print_blocking(string) -> None:
```
Comment (ancalita) [inline rasa/shared/utils/cli.py:9]: Is it possible to add a unit test for this new function?

**Review by ancalita — COMMENTED** [review]
Thanks for taking the initiative to fix this 💯 
To double-check, the ticket description referred to `2.x` minor series, is this occurring in `3.x` too?

**Comment by ancalita** [comment]
@trammell One more thing about `fcntl`, it seems this doesn't work on Windows, so we'll need to find a different solution for this OS...or a different cross-platform solution altogether.

**Comment by trammell** [comment]
@ancalita that is a very good point. I'm going to retract this PR and implement the changes discussed above.

**Pull request closed without merging**